simulation/traffic_simulator/include/traffic_simulator/math/catmull_rom_interface.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015-2022 TierIV.inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRAFFIC_SIMULATOR__MATH__CATMULL_ROM_INTERFACE_HPP_
+#define TRAFFIC_SIMULATOR__MATH__CATMULL_ROM_INTERFACE_HPP_
+
+#include <boost/optional.hpp>
+#include <exception>
+#include <geometry_msgs/msg/point.hpp>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace traffic_simulator
+{
+namespace math
+{
+class CatmullRomInterface
```

Comment:
I think it's naming is too ambiguous.
I think it is a base class for catmull rom spline curve.
How about rename to CatmullRomSplineBase or CatmullRomSplineInterface class.

Reply:
Ok, I renamed `CatmullRomInterface` to `CatmullRomSplineInterface`